src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdCaseUpdater.java

```diff
@@ -0,0 +1,205 @@
+    private final AuthTokenGenerator s2sTokenGenerator;
+    private final CoreCaseDataApi coreCaseDataApi;
+    private final ExceptionRecordFinalizer exceptionRecordFinalizer;
+
+    public CcdCaseUpdater(
+        CaseUpdateClient caseUpdateClient,
+        AuthTokenGenerator s2sTokenGenerator,
+        CoreCaseDataApi coreCaseDataApi,
+        ExceptionRecordFinalizer exceptionRecordFinalizer
+    ) {
+        this.caseUpdateClient = caseUpdateClient;
+        this.s2sTokenGenerator = s2sTokenGenerator;
+        this.coreCaseDataApi = coreCaseDataApi;
+        this.exceptionRecordFinalizer = exceptionRecordFinalizer;
+    }
+
+    @SuppressWarnings({"squid:S2139", "unchecked"}) // squid for exception handle + logging
+    public void updateCase(
+        ExceptionRecord exceptionRecord,
+        ServiceConfigItem configItem,
+        String idamToken,
+        String userId,
+        CaseDetails existingCase
+    ) {
+        log.info(
+            "Start updating case for {} with case ID {} from exception record {}",
+            configItem.getService(),
+            existingCase.getId(),
+            exceptionRecord.id
+        );
+
+        try {
+            String s2sToken = s2sTokenGenerator.generate();
+
+            SuccessfulUpdateResponse updateResponse = caseUpdateClient.updateCase(
+                configItem.getUpdateUrl(),
+                existingCase,
+                exceptionRecord,
+                s2sToken
+            );
+
+            log.info(
+                "Successfully updated case for {} with case ID {} from exception record {}",
+                configItem.getService(),
+                existingCase.getId(),
+                exceptionRecord.id
+            );
+
+            checkBulkScanReferenceIsSet(
+                (Map<String, ?>) updateResponse.caseDetails.caseData,
+                exceptionRecord.id
+            );
+
+            updateCaseInCcd(
+                idamToken,
+                s2sToken,
+                userId,
+                exceptionRecord,
+                updateResponse.caseDetails,
+                existingCase
+            );
+
+            log.info(
+                "Successfully updated case for {} with case ID {} from exception record {}",
+                configItem.getService(),
+                existingCase.getId(),
+                exceptionRecord.id
+            );
+        } catch (InvalidCaseDataException exception) {
+            throw new CallbackException("Failed to update case", exception);
+        } catch (PaymentsPublishingException exception) {
+            log.error(
+                "Failed to send update to payment processor for {} exception record {}",
+                configItem.getService(),
+                exceptionRecord.id,
+                exception
+            );
+
+            throw new CallbackException("Payment references cannot be processed. Please try again later", exception);
+        } catch (Exception exception) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['catch' branch identical to 'InvalidCaseDataException' branch](https://app.codacy.com/app/HMCTS/bulk-scan-orchestrator/pullRequest?prid=4558409)